tools/notification-configuration/notification-creator/Contacts.cs

```diff
@@ -109,14 +112,45 @@ private Uri GetCodeownersRepoUrl(BuildDefinition buildDefinition)
         return repoUrl;
     }
 
-
-    private CodeownersEntry GetMatchingCodeownersEntry(YamlProcess process, List<CodeownersEntry> codeownersEntries)
+    private CodeownersEntry GetMatchingCodeownersEntry(
+        YamlProcess process,
+        List<CodeownersEntry> codeownersEntries,
+        string repoUrl)
     {
         CodeownersEntry matchingCodeownersEntry =
-            CodeownersFile.GetMatchingCodeownersEntry(process.YamlFilename, codeownersEntries);
+            CodeownersFile.GetMatchingCodeownersEntry(
+                process.YamlFilename,
+                codeownersEntries,
+                UseRegexMatcher(repoUrl));
 
         matchingCodeownersEntry.ExcludeNonUserAliases();
 
         return matchingCodeownersEntry;
     }
+
+    /// <summary>
+    /// Whether the new regex-based CODEOWNERS matcher that supports wildcards should be used
+    /// for given repository. This method exists to allow incremental roll-out
+    /// of the new matcher [1].
+    ///
+    /// Note that enabling the new matcher will not change the reviewers assigned to
+    /// PRs, because this is handled by built-in GitHub matcher. 
+    /// But it will change who receives build failure notifications.
+    ///
+    /// [1] https://github.com/Azure/azure-sdk-tools/pull/5088
+    /// </summary>
+    private bool UseRegexMatcher(string repoUrl)
+    {
+        // Expected repoUrl: https://github.com/Azure/azure-sdk-for-net
+        if (repoUrl.Contains("azure-sdk-for-net"))
```

Comment:
Given this is only supposed to be temporary until we transition all the existing codeowners files I'm OK with this here but if we need to keep this longer I'm not sure I like hardcoding this here and we should consider some sort of config or option that gets passed through instead.

Reply:
@weshaggard got it. I will keep in mind the preferred way in general is to have proper options. As you noticed, this is temporary, so I will remove this code within few weeks max.